configuration_broker/config_broker/config_broker.h

```diff
@@ -0,0 +1,145 @@
+// Copyright Configured Things and CHERIoT Contributors.
+// SPDX-License-Identifier: MIT
+
+#include "cdefs.h"
+#include "compartment-macros.h"
+#include "token.h"
+#include <atomic>
+#include <compartment.h>
+#include <locks.hh>
+
+//
+// Internal representation of a token which allows an operation on
+// a configuration item.  This is put in a Capability sealed with one
+// of three different keys to define how it can be used.
+// Size and update_interval only used when setting the parser, but
+// keeping a common struct keeps the code a bit cleaner.
+//
+struct ConfigToken
+{
+	uint16_t   id;             // id for the capability, assigned on first use
+	size_t     size;           // size of the item
+	uint32_t   updateInterval; // Min MS between updates
+	const char Name[];         // Name of the configuration item
+};
+
+//
+// Create a Sealed Capability to read a config item
+//
+#define DEFINE_READ_CONFIG_CAPABILITY(name)                                    \
+                                                                               \
+	DECLARE_AND_DEFINE_STATIC_SEALED_VALUE(                                    \
+	  struct {                                                                 \
+		  uint16_t   id;                                                       \
+		  size_t     notUsed1;                                                 \
+		  uint32_t   notUsed2;                                                 \
```

Comment:
Not used because they're always initialised to zero but set later?  Or do they not need to be there?

Reply:
As per the preceding comment block its to keep the a common struct between all three capability types, so I can have a common unseal method in the broker.